packages/amplify-category-api/commands/api/add-graphql-datasource.js

```diff
@@ -205,3 +207,42 @@ async function getAwsClient(context, action) {
   const provider = require(providerPlugins[providerName]);
   return await provider.getConfiguredAWSClient(context, 'aurora-serverless', action);
 }
+
+function readSchema(projectDirectory, rdsSchema) {
+  const schemaFilePath = path.join(projectDirectory, 'schema.graphql');
+  const schemaDirectoryPath = path.join(projectDirectory, 'schema');
+  const schemaDirectoryExists = fs.existsSync(schemaDirectoryPath);
+  // convert the object schema to dump as file
+  const rdsSchemaDoc = mergeTypes([{}, rdsSchema], { all: true });
+
+  let schema;
+  if (schemaDirectoryExists) {
+    schema = readSchemaDocuments(schemaDirectoryPath).join('\n');
+    const rdsSchemaFilePath = path.join(schemaDirectoryPath, 'rds.graphql');
+    fs.writeFileSync(rdsSchemaFilePath, rdsSchemaDoc, 'utf8');
+  } else {
+    schema = fs.readFileSync(schemaFilePath).toString();
+  }
+  return schema;
+}
+
+function readSchemaDocuments(schemaDirectoryPath) {
+  const files = fs.readdirSync(schemaDirectoryPath);
+  let schemaDocuments = [];
+  for (const fileName of files) {
+    if (fileName.indexOf('.') === 0) {
+      continue;
+    }
+
+    const fullPath = `${schemaDirectoryPath}/${fileName}`;
+    const stats = fs.lstatSync(fullPath);
+    if (stats.isDirectory()) {
+      const childDocs = readSchemaDocuments(fullPath);
+      schemaDocuments = schemaDocuments.concat(childDocs);
+    } else if (stats.isFile()) {
+      const schemaDoc = fs.readFileSync(fullPath);
+      schemaDocuments.push(schemaDoc);
+    }
```

Comment:
Use globs to get all the files instead of traversing files and directory manually

Reply:
done!!